basxconnect/core/layouts/editperson/common/addresses.py

```diff
@@ -111,36 +110,45 @@ def edit_heading(model: type):
     return _("Edit %s") % pretty_modelname(model)
 
 
-def display_postal(postal: models.Postal):
+def display_postal():
+    postal = hg.C("i")
     modal = layout.modal.Modal.with_ajax_content(
         heading=edit_heading(models.Postal),
-        url=reverse_model(
+        url=ModelHref(
             models.Postal,
             "ajax_edit",
             kwargs={"pk": postal.pk},
             query={"asajax": True},
         ),
         submitlabel=_("Save"),
     )
-    is_inactive = postal.valid_until and postal.valid_until < timezone.now().date()
+    is_inactive = hg.F(
+        lambda c: c["i"].valid_until and c["i"].valid_until < timezone.now().date()
+    )
     return R(
         C(
             hg.DIV(
                 postal.type,
-                hg.BaseElement(" (", _("primary"), ")")
-                if postal.person.primary_postal_address
-                and postal.person.primary_postal_address.pk == postal.pk
-                else "",
+                hg.If(
+                    hg.F(
+                        lambda c: c["i"].person.primary_postal_address
+                        and c["i"].person.primary_postal_address.pk == c["i"].pk
+                    ),
+                    hg.BaseElement(" (", _("primary"), ")"),
+                ),
                 style="font-weight: bold; margin-bottom: 1rem;",
             ),
-            hg.DIV(postal.address, style="margin-bottom: 0.25rem;"),
+            hg.DIV(
+                ObjectFieldValue("address", object_contextname="i"),
+                style="margin-bottom: 0.25rem;",
+            ),
             hg.DIV(postal.postcode, " ", postal.city, style="margin-bottom: 0.25rem;"),
```

Comment:
here

Reply:
(etc)

Reply:
Ah, right. Nice. Btw. I also added support for item-accessor for things like ```hg.C("mydict")["key"]``` in one of the last commits. Maybe adding support for expressions or similar in the future.